api/src/main/java/com/velocitypowered/api/event/player/configuration/PlayerFinishConfigurationEvent.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2024 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.event.player.configuration;
+
+import com.velocitypowered.api.event.annotation.AwaitingEvent;
+import com.velocitypowered.api.proxy.Player;
+import org.jetbrains.annotations.NotNull;
+
+// TODO: Protocol State API
+/**
+ * This event is executed when the player is about to finish the Configuration state.
+ * <p>Velocity will wait for this event to finish the configuration phase on the client.</p>
+ *
+ * @param player The player who is about to complete the configuration phase
+ */
```

Comment:
You added the @since annotation to the PlayerEnterConfigurationEvent so you might want to add it here too.